examples/userapi/05_conditions.ipynb

```diff
@@ -0,0 +1,489 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ConditionalDimension tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial is designed to demonstrate how users can implement various conditions in Devito using [ Conditional Dimension](https://github.com/devitocodes/devito/blob/e918e26757a4f0ea3c3c416b6e0f48f4e5023c37/devito/types/dimension.py#L635 \"class ConditionalDimension(DerivedDimension):\"). Devito can generate a great variety of loop kernels using its Domain Specific Language. We are going to show how we can generate loop structures in C language using Devito's DSL and how we can add conditional statements in them.\n",
+    "\n",
+    "As always, we will begin by specifying some parameters. We will define a 2D iteration space and we will name the Dimensions as $i,j$.\n",
+    "\n",
+    "Initially we introduce a simple example of a Devito Operator and gently introduce the use of Conditional Dimension.\n",
```

Comment:
Carefully read through this entire section and see where you can improve. Here, add a little more description re. the 'plan'. Currently it's not clear where things are going or why for a few cells yet.